api/tests/mocha/migrations/extract-person-contacts.spec.js

```diff
@@ -5,81 +5,117 @@ const config = require('../../../src/config');
+      chai.expect(contactGet.callCount).to.equal(0);
       // No edits happened.
       chai.expect(insertDoc.callCount).to.equal(0);
       chai.expect(updatePlace.callCount).to.equal(0);
       chai.expect(createPerson.callCount).to.equal(0);
     });
   });
 
-  it('run attempts to revert the contact on error', done => {
-    // Migrate one type: Get the 3 levels of hierarchy
-    getView.onCall(0).callsArgWith(2, null, { rows: [ { id: 'a' }] });
-    getView.onCall(1).callsArgWith(2, null, { rows: [ ] });
-    getView.onCall(2).callsArgWith(2, null, { rows: [ ] });
+  it('run attempts to revert the contact on error', () => {
+    const facilityGenerator = async function* () {
+      yield 'a';
+    };
+    
+    const emptyGenerator = async function* () {
+    };
+    
+    // Only district_hospital has facilities, others are empty
+    contactGetUuids.onCall(0).returns(facilityGenerator());
+    contactGetUuids.onCall(1).returns(emptyGenerator());
+    contactGetUuids.onCall(2).returns(emptyGenerator());
 
-    // updateParents: Get doc for parent update
-    getDoc.onCall(0).callsArgWith(1, null, {
+    // Mock facility document for updateParents step
+    contactGet.onCall(0).resolves({
       _id: 'a',
       contact: { name: 'name', phone: 'phone'},
-      parent: {_id: 'b', contact: { name: 'name1', 'phone': 'phone1' }}
+      parent: {_id: 'b', contact: { name: 'name1', phone: 'phone1' }}
     });
-    // removeParent : insert place with deleted parent
-    insertDoc.onCall(0).callsArg(1);
+    
+    // removeParent: insert place with deleted parent
+    insertDoc.onCall(0).callsArgWith(1, null);
+    
     // resetParent: get parent to reset
-    getDoc.onCall(1).callsArgWith(1, null, { _id: 'b', contact: {}, newKey: 'newValue' });
-    // resetParent: update the place
+    contactGet.onCall(1).resolves({ 
+      _id: 'b', 
+      contact: {_id: 'existing-contact'}, 
+      newKey: 'newValue' 
+    });
+    
+    // resetParent: update the place - success
     updatePlace.onCall(0).resolves();
 
-    // Get doc for contact update
-    getDoc.onCall(2).callsArgWith(1, null, {
+    // Mock facility document for createPerson step (after parent update)
+    contactGet.onCall(2).resolves({
       _id: 'a',
-      contact: { name: 'name', phone: 'phone'},
-      parent: { _id: 'b', contact: {_id: 'f'}}
+      contact: { name: 'name', phone: 'phone'}
     });
-    // Update doc : deleted contact
-    insertDoc.onCall(1).callsArg(1);
-    // Create person doc
+    
+    // removeContact: delete contact from facility
+    insertDoc.onCall(1).callsArgWith(1, null);
+    
+    // createPerson: create person doc - success
     createPerson.onCall(0).resolves({ id: 'c'});
-    // Update doc : reset the contact field
-    updatePlace.onCall(1).returns(Promise.reject('error'));
-    // Restore the parent
+    
+    // resetContact: update doc with new contact - FAIL
+    updatePlace.onCall(1).rejects(new Error('error'));
```

Comment:
suggestion: can we revert this back to:

```js
    updatePlace.onCall(1).returns(Promise.reject('error'));
```

Then the error message we assert below should stay the same.